ng2-material/components/pagination/pagination.ts

```diff
@@ -0,0 +1,300 @@
+   * calculate range depending via model parameters
+   *
+   * @param {IPaginationModel} model
+   * @private
+   */
+  getRange() {
+    if (isPresent(this.model)) {
+      let rangeStart = (this.model.currentPage - 1) * this.model.itemsPerPage + 1;
+
+      let rest = this.model.totalItems - rangeStart,
+        rangeStop = rest < this.model.itemsPerPage ? this.model.totalItems : rangeStart + this.model.itemsPerPage - 1;
+
+      return this.getFormattedValue(rangeStart, rangeStop, this.model.totalItems);
+    }
+
+    return;
+  }
+}
+
+@Component({
+  selector: 'md-pagination-controls',
+  template: `
+      <span [class.md-pagination-control-active]="!isFirstPage()" class="md-pagination-control md-pagination-control-previous">
+        <button (click)="previousPage()" class="material-icons">keyboard_arrow_left</button>
+      </span>
+      <span [class.md-pagination-control-active]="!isLastPage()" class="md-pagination-control md-pagination-control-next">
+        <button (click)="nextPage()" class="material-icons">keyboard_arrow_right</button>
+      </span>
+    `,
+  host: {
+    '[class.md-pagination-controls]': 'true'
+  }
+})
+export class MdPaginationControls {
+  @Input()
+  name: string = 'default';
+
+  @Input()
+  model: IPaginationModel;
+
+  constructor(private service: PaginationService) {
+    this.service.onChange
+      .filter(event => isPresent(event) && isPresent(event.name))
+      .filter(event => event.target === this.name)
+      .subscribe(event => {
+        this.model = event.pagination;
+      });
+  }
+
+  isFirstPage() {
+    return isPresent(this.model) && this.model.currentPage == 1;
+  }
+
+  isLastPage() {
+    return isPresent(this.model) && this.model.totalItems <= this.model.currentPage * this.model.itemsPerPage;
+  }
+
+  previousPage() {
+    if (isPresent(this.model)) {
+      this.changePage(this.model.currentPage - 1);
+    }
+  }
+
+  nextPage() {
+    if (isPresent(this.model)) {
+      this.changePage(this.model.currentPage + 1);
+    }
+  }
+
+  changePage(newPage: number) {
+    let model = JSON.parse(JSON.stringify(this.model));
+    model.currentPage = newPage;
+    this.service.change(this.name, model);
+  }
+}
+
+@Component({
+  selector: 'md-pagination-items-per-page',
+  template: `
+   <span *ngIf="itemsPerPageBefore" class="md-pagination-items-per-page-label md-pagination-items-per-page-before">{{itemsPerPageBefore}}</span>
+   <select [(ngModel)]="model.itemsPerPage" (ngModelChange)="changePaginationLength($event)" class="md-pagination-length-select">
+      <option *ngFor="#length of itemsPerPageOptions" [value]="length">
+        {{length}}
+      </option>
+    </select>
+    <span *ngIf="itemsPerPageAfter" class="md-pagination-items-per-page-label md-pagination-items-per-page-after">{{itemsPerPageAfter}}</span>
+  `,
+  host: {
+    '[class.md-pagination-items-per-page]': 'true',
+    '[hidden]': '!itemsPerPageOptions.length'
+  }
+})
+export class MdPaginationItemsPerPage {
```

Comment:
What about `MdPageSize`? I think Pagination is too verbose (even if it's more correct).